components/script/dom/svgsvgelement.rs

```diff
@@ -105,8 +104,9 @@ impl LayoutSVGSVGElementHelpers for LayoutDom<SVGSVGElement> {
                 .get_attr_for_layout(&ns!(), &local_name!("height"));
             let width = width_attr.map_or(DEFAULT_WIDTH, |val| val.as_uint());
             let height = height_attr.map_or(DEFAULT_HEIGHT, |val| val.as_uint());
+            let window = SVG.global().as_window();
```

Comment:
Ah, I see - this won't work from the layout thread. Rather than creating the GL context from the data() method, can we do that from SVGSVGElement's constructor instead?